users/views.py

```diff
@@ -154,3 +157,29 @@ def get_form_kwargs(self: "AccountSettingsView") -> dict[str, Any]:
         kwargs = super().get_form_kwargs()
         kwargs["instance"] = self.request.user.usersettings
         return kwargs
+
+
+class AccountDeleteView(LoginRequiredMixin, View):
+    template_name = "users/delete_account.html"
+    success_url = reverse_lazy("home")
+
+    def get(self: "AccountDeleteView", request: HttpRequest) -> HttpResponse:
+        form = ConfirmDeleteAccountForm(user=request.user)
+        return render(request, self.template_name, {"form": form})
+
+    def post(self: "AccountDeleteView", request: HttpRequest) -> HttpResponse:
+        form = ConfirmDeleteAccountForm(user=request.user, data=request.POST)
+        if form.is_valid():
+            request.user.deactivate()
+            logout(request)
+            messages.warning(request, "Your account has been deactivated. ")
+            messages.success(
+                request,
+                f"You have {settings.ACCOUNT_EXPIRATION_TIME_IN_DAYS} days to reactivate it."
+                f"After this time, your account will be permanently deleted."
+                f"To reactivate you account, contact our support."
```

Comment:
Ahh that's why admin has the option to send a link to reactivate. 
Curious case:
At the last day of the reactivate period I send message to support to reactivate. They send me the link. But I click it the day later. So I can not reactivate account because the window has passed, right?
This is tricky edge case. Should the reactivation/activation link be valid for certain amount of time? Let's say 48 hours (and this should be stated in the  email)? Let's discuss that, and update clickup task later on, with the conclusion.